llvm/test/Feature/Repo/repo_pruning_dependents_1.ll

```diff
@@ -1,27 +1,41 @@
-; This testcase tests the program repository pruning pass.
+; This testcase tests the program repository pruning pass for the dependents.
+; The explanation of the dependents is given in the repo_pruning_dependents.ll,
+; Please refer to the file for details.
 ;
-; If functions 'A' and 'B' are both dependent on function 'C',
-; only add a single TicketNode of 'C' to the 'repo.tickets'
-; in order to avoid multiple compilation_members of function
-; 'C' in the compilation.
+; This test checks there is no redundant compilation members added into the compilation.
+; In the test, functions 'A' and 'B' are both dependent on function 'C'. During the pruning
+; pass, only add a single TicketNode of 'C' to the 'repo.tickets' in order to avoid multiple
+; compilation_members of function 'C' in the compilation.
 
 ; The testcase includes three steps:
 ; Step 1: Build the code and create the database 'clang.db' which contains all Tickets.
+;         In this step, the digest of function `C` is generated by the RepoObjectWriter
+;         and added into the dependent of the functions 'A' and `B`.
 ; Step 2: Dump all compilation members in the database;
+;         There are 3 compilation members, which are functions `A`, `B` and `C`.
 ; Step 3: Re-build the same IR code again.
+;         In this step, the IR code firstly pass the RepoHashGenerator pass to generate the
+;         digests of functions `A`, `B` and `C`. During the RepoPruning pass, the function `A`
+;         and `B` are pruned, only add a single TicketNode of their dependent `C` is added to
+;         the `repo.tickets`.
 ; Step 4: Dump all compilation members in the database again;
+;         The compilations, which are generated by step 1 and step 3, should be the same.
+;         Therefore, there are only one compilation, which inludes three members: functions
+;          `A`, `B` and `C`.
 ; Step 5: Check that the step 2 and step 4 generate the same compilation members.
-
-; This test only works for the Debug build because the digest of A is calculated for the Debug build.
-
+;
+; The test need to be updated if the compiler optimisation passes change.
+; To fix the test, run the step 3 to get the new digests of functions `A`
+; and `B`, then update the digests of functions `A` and `B` in the `TicketNode`
+; Metadata.
+;
 ; RUN: rm -f %t.db
 ; RUN: env REPOFILE=%t.db llc -mtriple="x86_64-pc-linux-gnu-repo" -filetype=obj %s -o %t
 ; RUN: env REPOFILE=%t.db pstore-dump  --all-compilations %t.db > %t.log
 ; RUN: env REPOFILE=%t.db clang -O3 -c --target=x86_64-pc-linux-gnu-repo -x ir %s  -o %t1
 ; RUN: env REPOFILE=%t.db pstore-dump  --all-compilations %t.db > %t1.log
 ; RUN: diff %t.log %t1.log
-
-; REQUIRES: asserts
```

Comment:
This test checks there is no redundant compilation members added into the compilation.